tracecat/cases/router.py

```diff
@@ -457,3 +471,445 @@ async def list_events_with_users(
     )
 
     return CaseEventsWithUsers(events=events, users=users)
+
+
+# Case Attachments
+
+
+@cases_router.get("/{case_id}/attachments")
+async def list_attachments(
+    *,
+    role: WorkspaceUser,
+    session: AsyncDBSession,
+    case_id: uuid.UUID,
+) -> list[CaseAttachmentRead]:
+    """List all attachments for a case."""
+    service = CasesService(session, role)
+    case = await service.get_case(case_id)
+    if case is None:
+        logger.warning("Case not found", case_id=case_id, user_id=role.user_id)
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Case with ID {case_id} not found",
+        )
+
+    attachments = await service.attachments.list_attachments(case)
+    return [
+        CaseAttachmentRead(
+            id=attachment.id,
+            case_id=attachment.case_id,
+            file_id=attachment.file_id,
+            file_name=attachment.file.name,
+            content_type=attachment.file.content_type,
+            size=attachment.file.size,
+            sha256=attachment.file.sha256,
+            created_at=attachment.created_at,
+            updated_at=attachment.updated_at,
+            creator_id=attachment.file.creator_id,
+            is_deleted=attachment.file.is_deleted,
+        )
+        for attachment in attachments
+    ]
+
+
+@cases_router.post("/{case_id}/attachments", status_code=status.HTTP_201_CREATED)
```

Comment:
it would be good to specify the shapes of each of the error details in the route metadata so openapi-ts can pick it up and give clients proper error types. e.g.
```python
@router.post(
    "/{repository_id}/sync",
    status_code=status.HTTP_204_NO_CONTENT,
    responses={
        422: {
            "model": RegistryRepositoryErrorDetail,
            "description": "Registry sync validation error",
        },
        404: {"description": "Registry repository not found"},
        400: {"description": "Cannot sync repository"},
    },
)
```